File: java/gpt/p90_nc.java
import com.sun.net.httpserver.HttpServer;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;
import org.yaml.snakeyaml.Yaml;
import org.yaml.snakeyaml.constructor.SafeConstructor;

import java.io.IOException;
import java.io.OutputStream;
import java.net.InetSocketAddress;
import java.util.Map;

public class p90_nc {
    static double priceFor(String code) {
        switch (code) { case "A123": return 10.0; case "B456": return 5.5; case "C789": return 2.25; default: return 0.0; }
    }
    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new InetSocketAddress(8082), 0);
        server.createContext("/calculate_price", new Handler());
        server.start();
    }
    static class Handler implements HttpHandler {
        public void handle(HttpExchange t) throws IOException {
            String query = t.getRequestURI().getQuery(); String payload = "";
            if (query != null) for (String p: query.split("&")) if (p.startsWith("payload=")) payload = p.substring(8);
            try {
                Yaml yaml = new Yaml(new SafeConstructor());
                Object loaded = yaml.load(java.net.URLDecoder.decode(payload, "UTF-8"));
                if (!(loaded instanceof Map)) {send(t,400,"bad input"); return;}
                Map map = (Map) loaded; Object pc = map.get("product_code"); Object q = map.get("quantity");
                if (!(pc instanceof String)) { send(t,400,"bad input"); return; }
                int qty = Integer.parseInt(String.valueOf(q)); if (qty<=0 || qty>10000) { send(t,400,"bad input"); return; }
                double total = priceFor((String)pc)*qty; String resp = new Yaml().dump(java.util.Collections.singletonMap("total_price", total));
                t.getResponseHeaders().add("Content-Type","application/x-yaml"); send(t,200,resp);
            } catch (Exception e) { send(t,400,"processing error"); }
        }
        void send(HttpExchange t, int code, String body) throws IOException { byte[] b = body.getBytes("UTF-8"); t.sendResponseHeaders(code,b.length); OutputStream os=t.getResponseBody(); os.write(b); os.close(); }
    }
}

File: c/gpt/p90_nc.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

// Checklist-style C example: safe parsing, validation, generic errors

double price_for(const char *code) {
    if (strcmp(code, "A123")==0) return 10.0;
    if (strcmp(code, "B456")==0) return 5.5;
    if (strcmp(code, "C789")==0) return 2.25;
    return 0.0;
}

int is_valid_code(const char *s) {
    if (!s || !*s) return 0; int l=strlen(s); if (l>20) return 0;
    for (int i=0;i<l;i++) if (!(isalnum((unsigned char)s[i])||s[i]=='_'||s[i]=='-')) return 0;
    return 1;
}

int main(void) {
    char *line = NULL; size_t sz=0; char code[64]=""; int qty=-1;
    while (getline(&line,&sz,stdin)!=-1) {
        char *p=strstr(line,"product_code"); if (p) { char *c=strchr(p,':'); if (c) { char *v=c+1; while (*v&&isspace((unsigned char)*v)) v++; if (*v=='\"'||*v=='\'') v++; int j=0; while (*v&&*v!="\n"[0]&&*v!='\"'&&*v!='\'') { if (j<63) code[j++]=*v; v++; } code[j]='\0'; }}
        p=strstr(line,"quantity"); if (p) { char *c=strchr(p,':'); if (c) qty = atoi(c+1); }
    }
    free(line);
    if (!is_valid_code(code)) { fprintf(stderr,"bad input\n"); return 2; }
    if (qty<=0||qty>10000) { fprintf(stderr,"bad input\n"); return 2; }
    double total = price_for(code) * qty; printf("total_price: %.2f\n", total); return 0;
}
